packages/api/src/router/widgets/stocks.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from "zod";
+
+import { fetchStockPriceHandler } from "@homarr/request-handler/stock-price";
+
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+const stockPriceInputSchema = z.object({
+  stock: z.string().nonempty(),
+  timeRange: z.string().nonempty(),
+  timeInterval: z.string().nonempty(),
```

Comment:
Since we only offer pre-defined values for these properties, we could likely create a type for this, no?
Example:
```ts
z.enum(["7d", "2w", "1M", "6M", "1y"])
```